### PR TITLE
refactor: enable go-critic

### PR DESCRIPTION
**Pull request by moldis**
Closes https://github.com/celestiaorg/celestia-app/issues/2987

- Added gocritics to linter
- Fixed minor gocritics linter errors

### CODE REVIEW

File: .golangci.yml [hunk 1]
```diff
@@ -12,6 +12,7 @@ linters:
     - prealloc
     - stylecheck
     - unconvert
+    - gocritic
```
Comment (rootulp) [inline .golangci.yml:15]: [no change needed] this list was so close to being sorted alphabetically lol

File: pkg/shares/shares_test.go [hunk 2]
```diff
@@ -17,19 +17,21 @@ func TestSequenceLen(t *testing.T) {
 		wantLen uint32
 		wantErr bool
 	}
-	sparseNamespaceID := bytes.Repeat([]byte{1}, appconsts.NamespaceSize)
-	firstShare := append(sparseNamespaceID,
+	firstShare := append(bytes.Repeat([]byte{1},
```
Comment (rootulp) [inline pkg/shares/shares_test.go:20]: golangci-lint is complaining here

File: x/upgrade/keeper.go [hunk 3]
```diff
@@ -29,11 +29,9 @@ type Fraction struct {
 // between 2/3 and 3/3 providing 1/6 fault tolerance to halting the
 // network during an upgrade period. It can be modified through a
 // hard fork change that modified the app version
-func SignalThreshold(version uint64) Fraction {
-	switch version {
-	default:
-		return defaultSignalTheshold
-	}
+func SignalThreshold(_ uint64) Fraction {
+	// TODO not using param in function header, return default.
```
Comment (rootulp) [inline x/upgrade/keeper.go:33]: [blocking] What is the function header? proposal to remove this comment

```suggestion
```

**Review by rootulp — CHANGES_REQUESTED** [review]
Thanks for the PR! It looks like golangci-lint is complaining still so I think a few more changes need to be made before we can merge this.

We're in a transitory phase where we're extracting a few of the files touched in this PR to https://github.com/celestiaorg/go-square. So if we enable go-critic here then we should also enable go-critic in that repo. Eventually the files in this repo that were moved to go-square will be removed.

File: x/upgrade/keeper.go (hunk 3 above)
Comment (moldis) [reply x/upgrade/keeper.go:33]: Removed

**Comment by rootulp** [comment]
@moldis is this ready for review? It looks like golangci-lint passes [here](https://github.com/celestiaorg/celestia-app/actions/runs/7528159886/job/20501848905?pr=3000).

**Review by rootulp — APPROVED** [review]
LGTM! Thanks for the contribution @moldis

**Review by staheri14 — APPROVED** [review]
Thanks, Looks good to me too!

**Review by ninabarbakadze — APPROVED** [review]
thank you!

**Review by evan-forbes — APPROVED** [review]

**Pull request merged**